mongo/integration/mtest/mongotest.go

```diff
@@ -497,6 +505,19 @@ func (t *T) createTestClient() {
 			t.failed = append(t.failed, cfe)
 		},
 	})
+	// only specify connection pool monitor if no deployment is given
+	if clientOpts.Deployment == nil {
```

Comment:
This check is here because we don't allow specifying topology, server, or connection options if the `Deployment` option is set.